tmail-backend/mailbox/encrypted/api/src/main/scala/com/linagora/tmail/encrypted/event/EncryptedProjectionCleanupListener.java

```diff
@@ -0,0 +1,58 @@
+package com.linagora.tmail.encrypted.event;
+
+import javax.inject.Inject;
+
+import org.apache.james.events.Event;
+import org.apache.james.events.EventListener.ReactiveGroupEventListener;
+import org.apache.james.events.Group;
+import org.apache.james.mailbox.events.MailboxEvents.Expunged;
+import org.apache.james.mailbox.model.MessageMetaData;
+import org.reactivestreams.Publisher;
+
+import com.linagora.tmail.encrypted.EncryptedEmailContentStore;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class EncryptedProjectionCleanupListener implements ReactiveGroupEventListener {
+
+    public static class EncryptedProjectionCleanupGroup extends Group {
+    }
+
+    static final Group GROUP = new EncryptedProjectionCleanupGroup();
+
+    private final EncryptedEmailContentStore encryptedEmailContentStore;
+
+    @Inject
+    public EncryptedProjectionCleanupListener(EncryptedEmailContentStore encryptedEmailContentStore) {
+        this.encryptedEmailContentStore = encryptedEmailContentStore;
+    }
+
+    @Override
+    public Group getDefaultGroup() {
+        return GROUP;
+    }
+
+    @Override
+    public boolean isHandling(Event event) {
+        return event instanceof Expunged;
+    }
+
+    @Override
+    public Publisher<Void> reactiveEvent(Event event) {
+        if (event instanceof Expunged) {
+            Expunged expunged = (Expunged) event;
+
+            return handleMessageDeletion(expunged);
+        }
+        return Mono.empty();
+    }
+
+    private Mono<Void> handleMessageDeletion(Expunged expunged) {
+        return Flux.fromIterable(expunged.getExpunged()
+            .values())
+            .map(MessageMetaData::getMessageId)
+            .flatMap(messageId -> Mono.from(encryptedEmailContentStore.delete(messageId)))
```

Comment:
We need to delete the underlying data only if the message is no longer accessible.

Overwize you might destroy it upon a move.

Can you add a test showing that move do not delete the GPG representation of the message?